basis/cairo/gobject/ffi/ffi.factor

```diff
@@ -0,0 +1,10 @@
+! Copyright (C) 2024 knottio.
+! See https://factorcode.org/license.txt for BSD license.
+USING: alien.syntax gobject-introspection vocabs ;
+IN: cairo.gobject.ffi
+
+<< "cairo.ffi" require >>
+
+LIBRARY: cairo-gobject
+
+GIR: vocab:gir/cairo-1.0.gir
```

Comment:
it looks like we manually define the cairo.ffi, are there words here that we should manually define instead of using the GIR?  Alternatively, should we only use the GIR?

Reply:
Oh, I forgot that when I wrote the `.gobject.ffi` vocabs, I intended on extending `gobject-introspection` to check for the `foreign="1"` attribute and handle them accordingly. But now that I looked at it again, that wouldn't work quite right, since some of the cairo structs needed by GDK4 are fully defined in the GIR rather than setting `foreign="1"`. So the current approach--as used by `gdk2`--of writing out the `FOREIGN-*-TYPE:` lines seems just fine. I thought there was a lot more to the cairo GIR than there is. Looks like only a few more types than what's defined in `gdk2.ffi`. I'll switch it over to that approach; that'll get rid of the double definitions I was seeing.